### PR TITLE
fix: Settings on 'Security' tab not persisting correctly after refactor

### LINKED ISSUE #2406: Security settings are not persisting correctly
**To Reproduce**
1. Go to Settings -> Security
2. Toggle settings
3. Reload
4. Settings are not persisted as expected

Likely regressed with recent functional refactor

### PR DESCRIPTION
**Pull request by tommoor**
closes #2406

### CODE REVIEW

File: app/scenes/Settings/Security.js [hunk 1]
```diff
@@ -18,39 +18,51 @@ function Security() {
...
+      case "guestSignin":
+        setGuestSignin(ev.target.checked);
+        break;
+      default:
+    }
+  }, []);
 
+  React.useEffect(() => {
+    async function save() {
       await auth.updateTeam({
```
Comment (tommoor) [inline app/scenes/Settings/Security.js:49]: Not possible to use the variables immediately after `setState` as they are not updated, moved to a `useEffect` hook instead

File: app/scenes/Settings/Security.js [hunk 2]
```diff
@@ -18,39 +18,51 @@ function Security() {
   const team = useCurrentTeam();
   const { t } = useTranslation();
   const { showToast } = useToasts();
-  const [sharing, setSharing] = useState(team.documentEmbeds);
-  const [documentEmbeds, setDocumentEmbeds] = useState(team.guestSignin);
-  const [guestSignin, setGuestSignin] = useState(team.sharing);
+  const [sharing, setSharing] = useState(team.sharing);
```
Comment (tommoor) [inline app/scenes/Settings/Security.js:21]: These were all mixed up.

File: app/scenes/Settings/Security.js [hunk 3]
```diff
@@ -18,39 +18,51 @@ function Security() {
...
         guestSignin,
       });
 
       showSuccessMessage();
-    },
-    [auth, sharing, documentEmbeds, guestSignin, showSuccessMessage]
-  );
+    }
+
+    if (
```
Comment (tommoor) [inline app/scenes/Settings/Security.js:58]: Not the most scalable solution, but it's okay for now

**Comment by iamsaumya** [comment]
Hey, I worked on this as well. 

If the approach looks good to you, I will push changes. [Gist](https://gist.github.com/iamsaumya/97f68dd6db1ac734c2181106815028f7)

**Comment by tommoor** [comment]
Combining the state makes a big difference, refactored based on that – I think this is the best of both now.

File: app/scenes/Settings/Security.js [hunk 4]
```diff
@@ -18,38 +18,29 @@ function Security() {
...
-          break;
-        case "documentEmbeds":
-          setDocumentEmbeds(ev.target.checked);
-          break;
-        case "guestSignin":
-          setGuestSignin(ev.target.checked);
-          break;
-        default:
-      }
+      const newData = { ...data, [ev.target.name]: ev.target.checked };
```
Comment (iamsaumya) [inline app/scenes/Settings/Security.js:36]: Is this how we set the key at runtime? 😮 Learned something new!

**Review by iamsaumya — APPROVED** [review]

**Pull request merged**